### PR TITLE
Source Google Ads: fix for date range too narrow exception

### LINKED ISSUE #4806: source google ads exception DATE_RANGE_TOO_NARROW
## Enviroment
- **Airbyte version**:  example is 0.27.2-alpha
- **OS Version / Instance**: macOS, Windows 7/10, Ubuntu 18.04, GCP n2. , AWS EC2
- **Source Connector and version**: source-google-ads v0.1.2
- **Severity**: Medium 
- **Step where error happened**:  Sync job

## Current Behavior
Start date and end date range not calculated correctly.  

## Expected Behavior
Start date should always be greater than end date.

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```
File "/usr/local/lib/python3.7/site-packages/google/ads/googleads/interceptors/exception_interceptor.py", line 141, in _handle_grpc_failure
    raise self._get_error_from_response(response)
google.ads.googleads.errors.GoogleAdsException: (<_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INVALID_ARGUMENT
	details = "Request contains an invalid argument."
	debug_error_string = "{"created":"@1626522941.925908324","description":"Error received from peer ipv4:142.250.183.170:443","file":"src/core/lib/surface/call.cc","file_line":1066,"grpc_message":"Request contains an invalid argument.","grpc_status":3}"
>, <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INVALID_ARGUMENT
	details = "Request contains an invalid argument."
	debug_error_string = "{"created":"@1626522941.925908324","description":"Error received from peer ipv4:142.250.183.170:443","file":"src/core/lib/surface/call.cc","file_line":1066,"grpc_message":"Request contains an invalid argument.","grpc_status":3}"
>, errors {
  error_code {
    query_error: DATE_RANGE_TOO_NARROW
  }
  message: "\'segments.date\' start with 2021-07-18 end at 2021-07-15 in WHERE clause must be constrained to a date range of at least 1 day."
}
```

</details>

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by blotouta2**
## What
Date Range Exception 

## How
Add support for start and end date calculation for accessing google ads report  
resolves https://github.com/airbytehq/airbyte/issues/4806

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def get_date_params(stream_slice: Mapping[str, Any], cursor_field: str, end_date
         if start_date > pendulum.now():
             return start_date.to_date_string(), start_date.add(days=1).to_date_string()
 
-        end_date = min(end_date, pendulum.parse(stream_slice.get(cursor_field)).add(months=1))
+        end_date = max(end_date, pendulum.parse(stream_slice.get(cursor_field)).add(months=1))
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:80]: If we accept your changes, then we will get a lot of unnecessary duplicates.
The code now works in such a way that we read data in one month.
For example, start_date is 2021-03-18, then we will have several requests:

| start_date   |      end_date      | 
|----------|:-------------:|
| 2021-03-18 |  2021-04-18 |
| 2021-04-19 |  2021-05-18 |
| 2021-05-19 |  2021-06-18 |
| 2021-06-19 |  2021-07-18 |
| 2021-07-19 |  2021-07-18 | 

**This is where the code breaks down**

If we well use your changes, then the following situation will be:
| start_date   |      end_date      | 
|----------|:-------------:|
| 2021-03-18 |  2021-07-18 |
| 2021-04-19 |  2021-07-18 |
| 2021-05-19 |  2021-07-18 |
| 2021-06-19 |  2021-07-18 |
| 2021-07-19 |  2021-08-18 | 

The problem exists, but change `min` to `max` is not the correct solution.

**Review by yevhenii-ldv — CHANGES_REQUESTED** [review]
Please looks comment.

**Comment by yevhenii-ldv** [comment]
Hi @blotouta2 
I have already added fix to #4788

**Comment by sherifnada** [comment]
@blotouta2 I think we should be good to go according to @yevhenii-ldv comments

**Pull request closed without merging**